File: TodoApi1.Frontend/src/models/todoItem.ts
export enum TodoSteps {
  Uncomplete,
  InProgress,
  Complete,
}

export enum Categories {
  NoCategory,
  Home,
  Work,
  Entertaiment,
}

export interface TodoItem {
  id: number;
  name: string;
  todoStep: TodoSteps;
  description: string;
  creationDate: Date;
  category: Categories;
  orderId: number;
}

File: TodoApi1.Backend/Models/TodoItem.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace TodoApi1.Models
{
    public enum TodoSteps
    {
        Uncomplete,
        InProgress,
        Complete
    }
    public enum Categories
    {
        NoCategory,
        Home,
        Work,
        Entertaiment
    }
    public class TodoItem
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public TodoSteps TodoStep { get; set; }
        public string Description { get; set; }
        public DateTime CreationDate { get; set; }
        public Categories Category { get; set; }
        public int OrderId { get; set; }
    }
}